bot/__init__.py

```diff
@@ -104,136 +239,33 @@ def _get_word(self) -> str:
-
-    # Check if a command in the form of `bot.tags['ask']`
-    # or alternatively `bot.tags['ask'] = 'whatever'` was used.
-    elif current == "[":
-        log.trace(f"Got a command candidate for getitem / setitem mimick: {self.buffer}")
-        # Syntax is `bot.tags['ask']` => mimic `getattr`
-        if self.buffer.endswith("]"):
-            # Key: The first argument, specified `bot.tags[here]`
-            key = self.buffer[self.index + 1:self.buffer.rfind("]")]
-            log.trace(f"Command mimicks getitem. Key: {key!r}")
-
-            # note: if not key, this corresponds to an empty argument
-            #       so this should throw / return a SyntaxError ?
-            args = ast.literal_eval(key)
-
-            # Use the cog's `get` command.
-            result = self.buffer[self.previous:self.index] + ".get"
-
-        # Syntax is `bot.tags['ask'] = 'whatever'` => mimic `setattr`
-        elif "=" in self.buffer and not self.buffer.endswith("="):
-            equals_pos = self.buffer.find("=")
-            closing_bracket_pos = self.buffer.rfind("]", 0, equals_pos)
-
-            # Key: The first argument, specified `bot.tags[here]`
-            key_contents = self.buffer[self.index + 1:closing_bracket_pos]
-            key = ast.literal_eval(key_contents)
-
-            # Value: The second argument, specified after the `=`
-            right_hand = self.buffer.split("=", maxsplit=1)[1].strip()
-            value = ast.literal_eval(right_hand)
-
-            # If the value is a falsy value - mimick `bot.tags.delete(key)`
-            if not value:
-                log.trace(f"Command mimicks delitem. Key: {key!r}.")
-                result = self.buffer[self.previous:self.index] + ".delete"
-                args = key
-
-            # Otherwise, assume assignment, for example `bot.tags['this'] = 'that'`
-            else:
-                # Allow using double quotes in triple double quote string
-                value = value.replace('"', '\\"')
-                log.trace(f"Command mimicks setitem. Key: {key!r}, value: {value!r}.")
-
-                # Use the cog's `set` command.
-                result = self.buffer[self.previous:self.index] + ".set"
-                args = f'"{key}" "{value}"'
-
-        # Syntax is god knows what, pass it along
-        # in the future, this should probably return / throw SyntaxError
-        else:
-            result = self.buffer
-            args = ''
-            log.trace(f"Command is of unknown syntax: {self.buffer}")
-
-        # Reconstruct valid discord.py syntax
-        self.buffer = f"{result} {args}"
-        self.index = len(result)
-        self.end = len(self.buffer)
-        log.trace(f"Mimicked command: {self.buffer}")
+    # If the command looks like a python syntax command, try to parse it.
+    if current == "(" or current == "[":
+        try:
+            python_result = parse_python(pos)
+            if python_result:
+                result = python_result
```

Comment:
hmm. maybe instead of checking for `None` (or is there another case that may occur?) here, wouldn't it be simpler and / or more intuitive to just `raise SyntaxError` within the parser?